terraform/getproviders/hash.go

```diff
@@ -0,0 +1,113 @@
+package getproviders
```

Comment:
Yes, we have. I improved the existing integration test.

https://github.com/gruntwork-io/terragrunt/blob/fb18445a052257517a6229d1921b864a20bd1b82/test/integration_test.go#L271

Here we check that the lock file has a _provider_ entry in the `.terraform.lock.hcl`.  
We also run all our integration tests twice, with `TERRAGRUNT_PROVIDER_CACHE=1` enabled and not, is also a  check for lock file generation.

Reply:
I only renamed `terraform/provider/hash.go` to `terraform/getproviders/hash.go`. In the last pull request, we discussed this, and I said that some of the code was taken from terraform 1.5, and since we are already using this package, we decided it was not worth adding anything. No new code from `terraform` repo was used for this pull request.

Reply:
Roger